### PR TITLE
Remove incorrect phrase from .NET 8 release notes

### PR DESCRIPTION
**Pull request by stephentoub**
## Summary

string.IndexOfAny doesn't have an overload for SearchValues and that's by design.

---

#### Internal previews

| 📄 File | 🔗 Preview link |
|:--|:--|
| [docs/core/whats-new/dotnet-8.md](https://github.com/dotnet/docs/blob/a001644ff29f8811522eebe36c9342097cb9031c/docs/core/whats-new/dotnet-8.md) | [What's new in .NET 8](https://review.learn.microsoft.com/en-us/dotnet/core/whats-new/dotnet-8?branch=pr-en-us-39084) |

### CODE REVIEW

File: docs/core/whats-new/dotnet-8.md [hunk 1]
```diff
@@ -481,7 +481,7 @@ IDataView predictions = model.Transform(split.TestSet);
...
+- The new <xref:System.Buffers.SearchValues%601?displayProperty=fullName> type is designed to be passed to methods that look for the first occurrence of any value in the passed collection. For example, <xref:System.String.IndexOfAny(System.Char[])?displayProperty=nameWithType> looks for the first occurrence of any character in the specified array in the `string` it's called on. NET 8 adds new overloads of methods like <xref:System.MemoryExtensions.IndexOfAny%2A?displayProperty=nameWithType> that accept an instance of the new type. When you create an instance of <xref:System.Buffers.SearchValues%601?displayProperty=fullName>, all the data that's necessary to optimize subsequent searches is derived *at that time*, meaning the work is done up front.
```
Comment (gewarren) [inline docs/core/whats-new/dotnet-8.md:484]:
```suggestion
- The new <xref:System.Buffers.SearchValues%601?displayProperty=fullName> type is designed to be passed to methods that look for the first occurrence of any value in the passed collection. For example, <xref:System.String.IndexOfAny(System.Char[])?displayProperty=nameWithType> looks for the first occurrence of any character in the specified array in the `string` it's called on. .NET 8 adds new overloads of methods like <xref:System.MemoryExtensions.IndexOfAny%2A?displayProperty=nameWithType> that accept an instance of the new type. When you create an instance of <xref:System.Buffers.SearchValues%601?displayProperty=fullName>, all the data that's necessary to optimize subsequent searches is derived *at that time*, meaning the work is done up front.
```

Comment (gewarren) [reply docs/core/whats-new/dotnet-8.md:484]: Also, does the sentence that starts "For example" need to use a different method as an example then?

**Review by gewarren — APPROVED** [review]

File: docs/core/whats-new/dotnet-8.md (hunk 1 above)
Comment (stephentoub) [reply docs/core/whats-new/dotnet-8.md:484]: It's technically not wrong, but it could be changed to refer to MemoryExtensions.IndexOfAny.

Comment (gewarren) [reply docs/core/whats-new/dotnet-8.md:484]: I updated my suggestion.

Comment (stephentoub) [reply docs/core/whats-new/dotnet-8.md]: Thanks

**Pull request merged**